src/com/google/enterprise/adaptor/prebuilt/PrebuiltTransforms.java

```diff
@@ -205,26 +287,61 @@ public String toString() {
 
   /** Contains source and destination metadata key. */
   private static class KeyPairing {
-    private final String src;
-    private final String dest;
+    private final Key src;
+    private final Key dest;
 
-    KeyPairing(String from, String to) {
+    KeyPairing(Key from, Key to) {
       if (null == from || null == to) {
         throw new NullPointerException();
       } 
       src = from;
       dest = to;
     }
 
+    @Override
+    public String toString() {
+      return "(from=" + src + ",to=" + dest + ")";
+    }
+  }
+
+  /** Contains a key in keyset. */
+  private static class Key {
+    private final String key;
+    private final Keyset keyset;
+
+    Key(String key, String keyset) {
+      this(key, Keyset.from(keyset));
+    }
+
+    Key(String key, Keyset keyset) {
+      if (key == null || keyset == null) {
+        throw new NullPointerException();
+      }
+      this.key = key;
+      this.keyset = keyset;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
You usually want to override hashCode if you override equals. I know this is a private class (now).